src/gluonts/dataset/schema.py

```diff
@@ -0,0 +1,160 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from dataclasses import dataclass
+from typing import Any, Dict, Optional, Union, Generic, TypeVar, Type
+
+import numpy as np
+import pandas as pd
+import pyarrow as pa
+from pandas.tseries.frequencies import to_offset
+
+from gluonts.exceptions import GluonTSDataError
+
+T = TypeVar("T")
+
+
+class FieldType(Generic[T]):
+    def __call__(self, v: Any) -> T:
+        raise NotImplementedError()
+
+    def __eq__(self, other):
+        raise NotImplementedError()
+
+    def is_compatible(self, v: Any) -> bool:
+        raise NotImplementedError()
+
+
+@dataclass
+class NumpyArrayField(FieldType[np.ndarray]):
+    dtype: Type = np.float32
+    ndim: Optional[int] = None
+
+    def __call__(self, value: Any) -> np.ndarray:
+        if isinstance(value, pa.Array):
+            value = value.to_numpy()
+            value = value.astype(self.dtype)
+        else:
+            value = np.asarray(value, dtype=self.dtype)
+        if self.ndim is not None and self.ndim != value.ndim:
```

Comment:
Let's add an empty line here.
```suggestion

        if self.ndim is not None and self.ndim != value.ndim:
```